package.json

```diff
@@ -5,20 +5,19 @@
     "build": "vite build",
     "lint": "eslint --cache .",
     "lint:fix": "npm run lint -- --fix",
-    "test": "vitest",
+    "test": "vitest run",
     "prepare": "husky install",
     "commit": "commit"
   },
-  "type": "module",
```

Comment:
Having converted the commitlint file to TS, it was not able to run because this package option means that every ts file in scope gets treated as a module. But the commitlint _is not a module_. 

Then having removed this option you'll notice the build output types have changed. This is expected, and not really an issue.

Reply:
Changed my mind on this. It's not worth compromising on the "module" nature of this template for a little linting convenience. I've added some changes to the lint config to make clearer how it should handle different file extensions.